### ISSUE TITLE
Вопросы касательно GNU GPL v3

### ISSUE THREAD

**Issue by plzombie** [ImageProcessing-ElectronicPublications/scantailor-experimental#9]
Раз мы разрабатываем под этой "свободной" лицензией, надо выполнить несколько требований:

* Заменить везде в программе Scan Tailor на Scan Tailor Experimental, чтобы было понятно, что это форк.
* Добавить кнопку "скачать исходники" в доки, которая бы вела на `https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/releases/tag/%VERSION%` (вообще, исходники в GNU GPL - это больная тема, особенно в 3-й версии. Там надо ещё и скрипты для сборки предоставлять, и используемые при сборке утилиты)

Ещё, у нас в проекте есть куча зависимостей, а у этих зависимостей есть свои лицензии. Их надо тоже, где-нибудь, через CMake запихнуть в один файлик с лицензией, который отображать в установщике.

Я знаю, что куча Open Source проектов ничего из этого не делают, но те же Google, Microsoft, Mozilla... У всех там война и мир в лицензионных соглашениях.

Если вы не против, я этим займусь в ближайшее время

**Comment by zvezdochiot** [comment]
@plzombie , делай. Я один чорт признаю только `public domain`. А во всём остальном я ни черта не понимаю.

**Comment by plzombie** [comment]
По лицензиям - сделал генерацию (надо вручную подавать папку, где будут файлики лицензий, потому что через cmake их не получить). Выглядит как-то так
![изображение](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/assets/1690651/9494f26d-8908-406d-9eda-10a10736a19d)
![изображение](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/assets/1690651/9fc1c965-6b3b-481c-b5a6-0c695538c429)
![изображение](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/assets/1690651/c0034aeb-2fb0-4d89-bae7-66233ff9fa93)

**Comment by plzombie** [comment]
[license_deps_dir.zip](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/files/13169350/license_deps_dir.zip)
Пример папки с файлами

**Comment by zvezdochiot** [comment]
@plzombie , может стоит их внести в новую папку `packaging/license`, чтоб не мучаться?

**Comment by plzombie** [comment]
Ну вот я собирал с libjpeg, там всего одну фразу в текст лицензии вставить о том, что используешь. А вот ты под линуксом, скорее всего, юзаешь libjpeg-turbo (да и не только ты), там целых 3 лицензии, включая оригинальную от libjpeg. И как в таком случае быть? Может, достаточно того, что я прикрепил пример папки? А потом нормальный скрипт установщика сделаем, который бы под виндой зависимости выкачивал, собирал и копировал куда что надо

**Comment by zvezdochiot** [comment]
@plzombie say:
>И как в таком случае быть?

Тогда должен быть доп. список соответсвия файла лицензии и зависимости. В зависимости от пользуемых зависимостей генерится список включаемых лицензий. Немного геморойно, но вариант рабочий.

**Comment by plzombie** [comment]
@zvezdochiot Да, согласен. И этим должен заниматься скрипт, а не мы руками прописывать

**Comment by zvezdochiot** [comment]
@plzombie , так я за то, что все файлы лицензий уже будут собраны, из их имён удалены все пробелы и прочая чертовщина, а местоположение более чем известно. Тогда у тебя не будет проблем с инструкциями `cmake`.
